bento_wes/db.py

```diff
@@ -198,97 +252,113 @@ def run_with_details_from_row(
                 run_id=run["id"],
                 state=run["state"],
                 request=run_request_from_row(run),
-                run_log=run_log_from_row(run, stream_content),
-                task_logs=cls.get_task_logs(c, run["id"]),
+                run_log=run_log_from_row(run, stream_content, self._settings),
+                task_logs=self.get_task_logs(self.c, run["id"]),
                 outputs=json.loads(run["outputs"]),
             )
         )
 
     @staticmethod
-    def _get_run_row(c: sqlite3.Cursor, run_id: uuid.UUID | str) -> sqlite3.Row | None:
+    def get_task_logs(c: sqlite3.Cursor, run_id: UUID | str) -> list:
+        c.execute("SELECT * FROM task_logs WHERE run_id = ?", (str(run_id),))
+        return [task_log_dict(task_log) for task_log in c.fetchall()]
+
+    @staticmethod
+    def _get_run_row(c: sqlite3.Cursor, run_id: UUID | str) -> sqlite3.Row | None:
         return c.execute("SELECT * FROM runs WHERE id = ?", (str(run_id),)).fetchone()
 
     @classmethod
-    def get_run(cls, c: sqlite3.Cursor, run_id: uuid.UUID | str) -> Run | None:
+    def get_run(cls, c: sqlite3.Cursor, run_id: UUID | str) -> Run | None:
         if run := cls._get_run_row(c, run_id):
             return run_from_row(run)
         return None
 
-    @classmethod
     def get_run_with_details(
-        cls,
-        c: sqlite3.Cursor,
-        run_id: uuid.UUID | str,
+        self,
+        run_id: UUID | str,
         stream_content: bool,
     ) -> RunWithDetails | None:
-        if run := cls._get_run_row(c, run_id):
-            return cls.run_with_details_from_row(c, run, stream_content)
+        if run := self._get_run_row(self.c, run_id):
+            return self.run_with_details_from_row(run, stream_content)
         return None
 
-    def set_run_log_name(self, run: Run, workflow_name: str):
-        # TODO: To avoid having multiple names, we should maybe only set this once?
-        self.cursor().execute("UPDATE runs SET run_log__name = ? WHERE id = ?", (workflow_name, run.run_id))
+    def set_run_log_name(self, run: Run, workflow_name: str) -> None:
+        self.c.execute(
+            "UPDATE runs SET run_log__name = ? WHERE id = ?",
+            (workflow_name, run.run_id),
+        )
         self.commit()
 
-    def set_run_log_command_and_celery_id(self, run: Run, cmd: Command, celery_id: int):
-        self.cursor().execute(
+    def set_run_log_command_and_celery_id(self, run: Run, cmd: Command, celery_id: int) -> None:
+        self.c.execute(
             "UPDATE runs SET run_log__cmd = ?, run_log__celery_id = ? WHERE id = ?",
-            (" ".join(cmd), celery_id, run.run_id),
+            (" ".join(shlex.quote(str(x)) for x in cmd), celery_id, run.run_id),
```

Comment:
q: what did this address?

Reply:
I ran into some issues that i dont quite remember with arguments not being recognized correctly due to spaces so used this to put them in quotes.